ksql-cli/src/main/java/io/confluent/ksql/util/TabularRow.java

```diff
@@ -111,14 +120,31 @@ private static void formatRow(
     for (int row = 0; row < numRows; row++) {
       builder.append('|');
       for (int col = 0; col < columns.size(); col++) {
-        builder.append(columns.get(col).get(row));
+        final String colValue = columns.get(col).get(row);
+        if (shouldClip(columns.get(col), numRows)) {
+          builder.append(colValue, 0, colValue.length() - CLIPPED.length())
+              .append(CLIPPED)
+              .append('|');
+        } else {
+          builder.append(colValue)
+              .append('|');
+        }
       }
       if (row != numRows - 1) {
         builder.append('\n');
       }
     }
   }
 
+  private static boolean shouldClip(final List<String> parts, final int rowsToPrint) {
+    // clip if there are more than one line and any of the remaining lines are non-empty
+    return parts.size() > rowsToPrint
+        && parts.subList(rowsToPrint, parts.size())
+        .stream()
+        .map(String::trim)
+        .noneMatch(String::isEmpty);
```

Comment:
> // clip if there are more than one line and any of the remaining lines are non-empty

Isn't this check "if there are more than one line and NO LINES ARE EMPTY".  So it would return `false` if there are multiple lines and one is empty.

I think maybe you want '! parts....allMatch(String::isEmpty)'

Reply:
good catch! added a test for it